examples/features/nse-composition/README.md

```diff
@@ -0,0 +1,371 @@
+${VOLUME_PATCH}
+patches:
+- target:
+    kind: Deployment
+    name: nse-firewall-vpp
+  patch: |-  
+    - op: replace  
+      path: /metadata/name
+      value: ${NAME}
+EOF
+if((i == 4))
+then
+cat > config-patch.yaml <<EOF
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nse-firewall-vpp
+spec:
+  template:    
+    spec:
+      containers:
+        - name: nse  
+          volumeMounts:
+            - mountPath: /etc/vppagent-firewall/config.yaml
+              subPath: config.yaml
+              name: vppagent-firewall-config-volume
+      volumes:
+        - name: vppagent-firewall-config-volume
+          configMap:
+            name: vppagent-firewall-config-file
+EOF
+fi
+  cat > patch-nse-firewall-vpp.yaml <<EOF
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nse-firewall-vpp
+spec:
+  template:    
+    spec:
+      containers:
+        - name: nse  
+          env:
+            - name: NSM_SERVICE_NAME
+              value: "nse-composition"
+            - name: NSM_LABELS
+              value: ${LABELS}
+      nodeSelector:  
+        kubernetes.io/hostname: ${NODE}    
+EOF
+  PASS_CFG="${PASS_CFG}- ./${f}"
+  if (( i < PASS_COUNT ))
+  then
+    PASS_CFG+=$'\n'
+  fi
+  cd ../
+done
+```
+
+Create customization file:
+```bash
+cat > kustomization.yaml <<EOF
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: ${NAMESPACE}
+
+bases:
+- ../../../apps/nsc-kernel
+- ../../../apps/nse-kernel
+${PASS_CFG}
+
+patchesStrategicMerge:
+- patch-nsc.yaml
+- patch-nse.yaml
+EOF
+```
+
+Create NSC patch:
+```bash
+cat > patch-nsc.yaml <<EOF
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nsc-kernel
+spec:
+  template:
+    spec:
+      containers:
+        - name: nsc
+          env:
+            - name: NSM_NETWORK_SERVICES
+              value: kernel://nse-composition/nsm-1
+      nodeSelector:
+        kubernetes.io/hostname: ${NODE}
+EOF
+```
+
+Create NSE patch:
+```bash
+cat > patch-nse.yaml <<EOF
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nse-kernel
+spec:
+  template:
+    spec:
+      containers:    
+        - name: nse
+          env:   
+            - name: NSM_CIDR_PREFIX
+              value: 172.16.1.100/31    
+            - name: NSM_SERVICE_NAME
+              value: "nse-composition"  
+            - name: NSM_REGISTER_SERVICE
+              value: "false"  
+            - name: NSM_LABELS
+              value: "app:gateway"
+        - name: nginx
+          image: networkservicemesh/nginx   
+          imagePullPolicy: IfNotPresent
+      nodeSelector:  
+        kubernetes.io/hostname: ${NODE}
+EOF                                                                                        
+```
+
+Create ConfigMap
+```bash
+cat > config-file.yaml <<EOF
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: vppagent-firewall-config-file
+  namespace: ${NAMESPACE}
+data:
+  config.yaml: |
+    aclrules:
+      allow icmp:
```

Comment:
Why do we need to initialize zero fields in config?

Reply:
fixed